tests/foreman/ui/test_products.py

```diff
@@ -220,13 +207,18 @@ def test_remove_prd(self, prd_name):
 
         """
         with Session(self.browser) as session:
-            make_product(
-                session,
-                org=self.organization.name,
-                loc=self.loc.name,
-                name=prd_name,
-                description=gen_string('alphanumeric'),
-            )
-            self.assertIsNotNone(self.products.search(prd_name))
-            self.products.delete(prd_name)
-            self.assertIsNone(self.products.search(prd_name))
+            for prd_name in generate_strings_list():
+                with self.subTest(prd_name):
+                    make_product(
+                        session,
+                        org=self.organization.name,
+                        loc=self.loc.name,
+                        name=prd_name,
+                        description=gen_string('alphanumeric'),
+                    )
+                    self.assertIsNotNone(self.products.search(prd_name))
+                    self.products.delete(prd_name)
+                    # Note: refresh is used here because sometimes selenium
+                    # is too fast to check the deleted object and it fails
+                    self.browser.refresh()
```

Comment:
Right, that is out of scope of that PR. We still need to implement general approach for katello-like (product, gpgkey, content views and etc) entities delete operations (as far as I remember @abalakh should look on that). Anyway, in final result, we have to improve `delete_entity` method or create similar one for situation like that.

If @sthirugn approach is stable enough, so leave it as it is as we should re-factor that part anyway.